File: sort.py
array = [1, 5, 2, 6, 3, 9, 7, 8]
#array   0, 1, 2, 3, 4, 5, 6, 7

def sort(x):
    for i in range(0, len(x)):
        min = i
        for j in range(i+1, len(x)):
            if x[min] > x[j]:
                min = j
        x[i], x[min] = x[min], x[i]
        print(i)
    return x

l = sort(array)
print(l)

File: sort.rs
fn main() {
    let mut x: [u8; 7] = [20, 10, 25, 15, 88, 41, 66];
    for n in 0..x.len() {
        let mut min = n;
        let a:u8 = x[n];
        for j in n+1..x.len() {
            if x[min] > x[j] {
                min = j;
            }
        }
        x[n] = x[min];
        x[min] = a;
    }
    print!("{:#?}", x);
}


        // x = [15, 30, 23, 22, 33] -> x = [23, 30, 15, 22, 33]
        // a = 3  b = 5
        // c = b                n = x[0]  x[n] = 10, x[min] = 10
        // b = a                x[0] = x[2]
        // a = c                x[2] = n
        // b = a                